### ISSUE TITLE
Improve logging for up/down events

### ISSUE THREAD

**Issue by bmfmancini** [Cacti/cacti#3509]
Hey Guys 

One thing that would be good to see is a new log for Up/Down events 
currently, when a device goes down it shows up in the error log which can be misleading thinking the system has tons of errors  I believe cacti related errors should go in the cacti.log and a new log i.e device_status should be created that hold these events to make log searching more clear 

Thanks !

**Comment by netniV** [comment]
I kinda agree, kinda disagree.  If you look at the installer, I have separate sections for different areas of the installer all going to install-*.log when it's enabled.  However, the only log file I ever ask for is install-complete.log which has everything.

So, I think if you want that separate log, it should go in both places but there would need to be again some house keeping for that.  Under OS packages, debian-based and redhat-based systems tend to use logrotate, so the packagers would need to be notified, and we would also have to create our own inhouse entries for when people use our sources rather than OS packages.

This then has the best of both worlds.  You can see the status' via the entire log (I have customers who continually check for green logs only!), and you can also get the "clean" version for the status log in its own right.

**Comment by bmfmancini** [comment]
Interesting point never considered the package level dependencies

**Comment by bmfmancini** [comment]
yeah looking at it more what's missing is when you filter the error section you will see the site down but not the restore stamp in the same view you would have to switch views so if we have a different category/file you would see all the pertinent info for that sites up/down timestamp in the same view 
this makes it easier I think to see all the events in one view will also make it easier for exports to a csv for reporting.

**Comment by TheWitness** [comment]
All those logs are included in the Monitor plugin.  Next one to merge into Core I'm guessing (along with Notification Lists).

**Comment by netniV** [comment]
Parts of monitor are.  But basic functionality is not.

**Comment by bmfmancini** [comment]
I think that monitor should be part of core the functions it has are what you would expect from a monitoring system out of the box without adding an additional plugin my opinion of course 
I feel the same way about THOLD as well but I know that adds to the complexity of the code base and issue tracking all from one repo could be messy 

but from a feature set monitor and thold are what you would expcect out of the box

**Comment by netniV** [comment]
Technically, it's a graphing system not a monitoring system but hard to tell that these days 👍

**Comment by bmfmancini** [comment]
Yea I think cacti has grown passed just a graphing system I would say the vast majority of users use cacti to monitor their devices rather than just get a graph 

Hope your keeping well

**Comment by cigamit** [comment]
Ya, I don't want to merge Thold in as is, as I was planning a complete rewrite of it.  Long ago I had started it, so we had completely dynamic alerts with escalation paths (and other plugins could add their own alert types), but I never finished.  With the current world situation, we will see if I have a bit of extra free time to start on it.

**Comment by cigamit** [comment]
Monitor itself was just suppose to be a display of Up / Down with audio alert.  I think any alerting on up / down in it should be striped out and added to the core (along with notification lists).  Leave Monitor itself out of the core though.

**Comment by netniV** [comment]
That sounds reasonable.

**Comment by TheWitness** [comment]
Agreeing whole hartedly.
